cfgov/research_reports/models.py

```diff
@@ -0,0 +1,235 @@
+                'body': subsection.body,
+                'url': format('#section-{}.{}', i, j),
+                'numbering': '' if is_appendix else format('{}.{} ', i, j),
+                'children': [{
+                    'title': section3.header,
+                    'body': section3.body,
+                    'url': format('#section-{}.{}.{}', i, j, k),
+                    'numbering': '' if is_appendix else
+                                 format('{}.{}.{} ', i, j, k),
+                } for k, section3 in enumerate(
+                    subsection.report_sections_level_three.all().order_by('pk')
+                )]
+            } for j, subsection in enumerate(
+                section.report_subsections.all().order_by('pk')
+            )]
+        } for i, section in enumerate(sections)]
+
+    return sec
+
+
+get_report_sections = get_report_parts()
+get_report_appendices = get_report_parts(True)
+
+
+def get_researchers():
+    return dict([
+        (r.title, r.url) for r in
+        SublandingFilterablePage.objects.get(pk=4833).get_children()
+    ])
+
+
+def _get_deploy_environment():
+    return getattr(settings, 'DEPLOY_ENVIRONMENT', 'local')
+
+
+class ReportSection(ClusterableModel):
+    header = models.CharField(max_length=200, blank=True)
+    is_appendix = models.BooleanField(default=False)
+    body = models.TextField(blank=True)
+    panels = [
+        FieldPanel('header'),
+        FieldPanel('is_appendix'),
+        FieldPanel('body'),
+        InlinePanel('report_subsections', label='Subsection'),
+    ]
+    report = ParentalKey('ResearchReportPage',
+                         on_delete=models.CASCADE,
+                         related_name='report_sections')
+
+
+class ReportSubSection(ClusterableModel):
+    header = models.CharField(max_length=200)
+    body = models.TextField(blank=True)
+    panels = [
+        FieldPanel('header'),
+        FieldPanel('body'),
+        InlinePanel(
+            'report_sections_level_three',
+            label='Section Level Three'
+        ),
+    ]
+    section = ParentalKey('ReportSection',
+                          on_delete=models.CASCADE,
+                          related_name='report_subsections')
+
+
+class ReportSectionLevelThree(models.Model):
+    header = models.CharField(max_length=200)
+    body = models.TextField(blank=True)
+    subsection = ParentalKey('ReportSubSection',
+                          on_delete=models.CASCADE,
+                          related_name='report_sections_level_three')
+
+
+class ReportAuthor(models.Model):
+    name = models.CharField(max_length=50)
+    report = ParentalKey('ResearchReportPage',
+                         on_delete=models.CASCADE,
+                         related_name='report_authors')
+
+
+class ReportMetaclass(WagtailAdminModelFormMetaclass):
+    @classmethod
+    def child_form(cls):
+        return ReportForm
+
+
+class ReportForm(WagtailAdminPageForm, metaclass=ReportMetaclass):
+    pass
```

Comment:
Because this form class hasn't been customized at all, are these needed?

Reply:
This was Wyatt's response when I asked him about that: https://github.com/cfpb/cfgov-refresh/pull/5767/files#r436171282

Reply:
Oooh. Past me was a little too optimistic. It turns out that, even with that workaround, nested inline panel support is still incomplete and doesn't always work as expected. For example, you can manually create nested inline items, but subsequently reordering them won't work. There are [a few open Wagtail issues](https://github.com/wagtail/wagtail/search?q=nested+inlinepanel&type=Issues) about this. For your purposes this might not matter if you are creating the items programmatically once and then never altering them manually.

Reply:
@wpears Is this code still doing what you intended? I couldn't discern a difference in how the nested panels work when it's removed. And we're on Wagtail 2.9 now, so that comment that references Wagtail 2.8+ should no longer apply.

Reply:
Yeah, that's probably vestigial at this point, if you've found that things still work as expected with it removed, let's remove it.